src/service.ts

```diff
@@ -1,116 +1,65 @@
-            value => task.of(successResponse(StatusCodes.OK)(value))
-        )
-    )
-
-export const getTransactionsBySnapshotHandler = (event: APIGatewayEvent, es: Client) =>
-    pipe(
-        taskEither.of<ApplicationError, APIGatewayEvent>(event),
-        chain(validateTransactionsEvent),
-        map(extractParameters),
-        chain(({term, limit, searchAfter}) => getTransactionBySnapshot(es)(term, limit, searchAfter)),
+        chain(validateAddressesEvent),
+        map(extractTerm),
+        chain(({ termName, termValue }) => getSnapshot(os)(termName, termValue)),
         fold(
             reason => task.of(errorResponse(reason)),
             value => task.of(successResponse(StatusCodes.OK)(value))
         )
     )
 
-export const getTransactionsByAddressHandler = (event: APIGatewayEvent, es: Client) =>
+export const getGlobalSnapshotRewards = (event: APIGatewayEvent, os: Client) =>
     pipe(
         taskEither.of<ApplicationError, APIGatewayEvent>(event),
         chain(validateAddressesEvent),
-        map(extractParameters),
-        chain(({term, limit, searchAfter}) => getTransactionByAddress(es)(term, null, limit, searchAfter)),
+        map(extractTerm),
+        chain(({ termName, termValue }) => getSnapshotRewards(os)(termName, termValue)),
         fold(
             reason => task.of(errorResponse(reason)),
             value => task.of(successResponse(StatusCodes.OK)(value))
         )
     )
 
-export const getTransactionsBySenderHandler = (event: APIGatewayEvent, es: Client) =>
+export const getBalanceByAddressHandler = (event: APIGatewayEvent, os: Client) =>
     pipe(
         taskEither.of<ApplicationError, APIGatewayEvent>(event),
         chain(validateAddressesEvent),
-        map(extractParameters),
-        chain(({term, limit, searchAfter}) => getTransactionBySender(es)(term, limit, searchAfter)),
+        map(extractOrdinal),
+        chain(({ term, ordinal }) => getBalanceByAddress(os)(term, ordinal)),
         fold(
             reason => task.of(errorResponse(reason)),
             value => task.of(successResponse(StatusCodes.OK)(value))
         )
     )
 
-export const getTransactionsByReceiverHandler = (event: APIGatewayEvent, es: Client) =>
-    pipe(
-        taskEither.of<ApplicationError, APIGatewayEvent>(event),
-        chain(validateAddressesEvent),
-        map(extractParameters),
-        chain(({term, limit, searchAfter}) => getTransactionByReceiver(es)(term, limit, searchAfter)),
-        fold(
-            reason => task.of(errorResponse(reason)),
-            value => task.of(successResponse(StatusCodes.OK)(value))
-        )
-    )
\ No newline at end of file
+const extractOrdinal = (event: APIGatewayEvent) => {
+    return {
+        term: event.pathParameters!.term,
+        ordinal: (event.queryStringParameters && event.queryStringParameters!.ordinal) || undefined
+    }
+}
+
+const extractTerm = (event: APIGatewayEvent) => {
+    if (event.pathParameters!.term == 'latest')
```

Comment:
Can we assume that `pathParameters` always exist? Because otherwise it would be more reasonable to use optional chaining `event.pathParameters?.term` instead of non-null assertion.

Reply:
This method should be called after validation, so `pathParameters` should be there. Maybe validation should also extract?

Reply:
Ok, fine then :)